File: programmers/42862/42862.py
def solution(n, lost, reserve):
    lost = set(lost)
    reserve = set(reserve)
    inter = lost & reserve
    lost -= inter
    reserve -= inter
    for l in sorted(lost) :
        if l-1 in reserve :
            reserve.remove(l-1)
            lost.remove(l)
            continue
        elif l+1 in reserve :
            reserve.remove(l+1)
            lost.remove(l)
            continue
            
    return n - len(lost)

File: C++/programmers/42862/42862.cpp
// #include <bits/stdc++.h>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> students(n+1, 1);
    
    for(int i = 0 ; i < lost.size() ; ++i) {
        students[lost[i]]--;
    }
    for(int i = 0 ; i < reserve.size() ; ++i) {
        students[reserve[i]]++;
    }
    
    for(int i = 1 ; i < students.size() ; ++i) {
        if (!students[i]) {
            if (i > 1 && students[i-1] > 1) {
                students[i-1]--;
                students[i]++;
            }
            else if (i < n && students[i+1] > 1) {
                students[i+1]--;
                students[i]++;
            }
        }
    }
    
    for (int i = 1 ; i < students.size() ; ++i) {
        if (students[i] >= 1)
            answer++;
    }
    
    return answer;
}